flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarFunctionCallGen.scala

```diff
@@ -48,10 +48,16 @@ class ScalarFunctionCallGen(
       .getOrElse(throw new CodeGenException("No matching signature found."))
     val resultClass = getResultTypeClass(scalarFunction, matchingSignature)
 
+    // zip for variable signatures
+    var paramToOperands = matchingSignature.zip(operands)
+    var i = paramToOperands.length
+    while (i < operands.length) {
+      paramToOperands = paramToOperands :+ (matchingSignature.head, operands(i))
```

Comment:
Thanks, @wuchong .Yes. It's a mistake here. And tests haven't covered this situation. Since the max number of the arguments is 254. I don't think it is necessary to use a component type at the phase of code generation. I will try to add some tests to cover this situation.